jsonschema2pojo-gradle-plugin/src/main/groovy/org/jsonschema2pojo/gradle/GenerateJsonSchemaAndroidTask.groovy

```diff
@@ -57,4 +59,12 @@ class GenerateJsonSchemaAndroidTask extends SourceTask {
 
     Jsonschema2Pojo.generate(configuration, new GradleRuleLogger(logger))
   }
+
+  void setTargetVersion(JsonSchemaExtension configuration) {
+    if (!configuration.targetVersion) {
+      def compileJavaTask = project.getTasksByName("compileJava", false).first()
```

Comment:
Could this be the cause for [actions failure](https://github.com/joelittlejohn/jsonschema2pojo/actions/runs/4206974193/jobs/7301154742)

>  Execution failed for task ':app:generateJsonSchema2PojoForDebug'.
> Cannot access first() element from an empty Iterable

Neither gradle nor android expert, but could it be that something similar to
```groovy
  void setTargetVersion(JsonSchemaExtension configuration) {
    if (!configuration.targetVersion) {
      configuration.targetVersion = project.android.compileOptions.sourceCompatibility
      logger.info 'Using Gradle sourceCompatibility as targetVersion for jsonschema2pojo: ' + configuration.targetVersion
    }
```
should be used for android (see [Android Gradle plugin API reference: CompileOptions](https://developer.android.com/reference/tools/gradle-api/7.4/com/android/build/api/dsl/CompileOptions) )?

Reply:
Yes, clearly the way the detection is working here is not right.

I have replicated the problem locally and I guess a combination of using a different property, and some more defensive coding, would be best.

Reply:
I don't have time to look this evening but can hopefully fix tomorrow (and release 1.2.1).

Reply:
We also need to get these CI jobs running against HEAD, not the latest release! :)

Reply:
> I don't have time to look this evening but can hopefully fix tomorrow (and release 1.2.1).

Whilst error is certainly not a pleasant one, there might be several workarounds (maybe not the nicest ones) eg.: provide target version in configuration, create/register a dummy `compileJava` task, appy `java` plugin.

I don't know whether it is a critical/big pain point for android developers or not, if not - perhaps it would make sense to take some time and not rush ?